src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsNavigationSource.cs

```diff
@@ -127,7 +127,7 @@ public IEnumerable<IEdmNavigationPropertyBinding> FindNavigationPropertyBindings
         {
             if (!navigationProperty.ContainsTarget)
             {
-                return this.NavigationPropertyBindings.Where(targetMapping => targetMapping.NavigationProperty == navigationProperty).ToList();
+                return this.NavigationPropertyBindings.Where(targetMapping => targetMapping.NavigationProperty == navigationProperty);
             }

```

Comment:
if this gets at all hot, it would be a good candidate for caching. you could iterate the navigationpropertybindings once, creating a dictionary of navigationProperty to NavigationPropertyBinding, and then just return the value from the dictionary each request.